src/tlsa.rs

```diff
@@ -0,0 +1,120 @@
+//! This types in this module relate to the TLSA and DANE
+//! as described by [RFC6698](https://datatracker.ietf.org/doc/html/rfc6698)
+use crate::Der;
+
+/// Corresponds to the TLSA DNS RR Type as defined by
+/// [RFC6698](https://datatracker.ietf.org/doc/html/rfc6698)
+#[derive(Clone, Debug, PartialEq)]
+pub struct TlsaRecord<'a> {
+    /// Specifies the provided association that will be used to match
+    /// the certificate presented in the TLS handshake.
+    pub certificate_usage: CertificateUsage,
+
+    /// specifies which part of the TLS certificate presented by the
+    /// server will be matched against the association data
+    pub selector: Selector,
+
+    /// Specifies how the certificate association is presented.
+    pub matching: MatchingType,
+
+    /// This field specifies the "certificate association data" to be
+    /// matched.  These bytes are either raw data (that is, the full
+    /// certificate or its SubjectPublicKeyInfo, depending on the selector)
+    /// for matching type 0, or the hash of the raw data for matching types 1
+    /// and 2.  The data refers to the certificate in the association, not to
+    /// the TLS ASN.1 Certificate object.
+    pub association_data: Der<'a>,
```

Comment:
As you alluded to, I don't think this is always `Der<'_>`, at least I assume that in the hash cases these would just be straight up bytes?

How do you feel about modeling it more like this?

```rust
enum CertificateUsage {
    CaConstraint(Selector),
    ..
}

enum Selector {
    FullCertificate(MatchingType),
}

enum MatchingType {
     Exact(Der<'_>),
     Sha256([u8; 32]),
     Sha512([u8; 64]),
}
```